### PR TITLE
Redux toolkit + mock backend

### PR DESCRIPTION
**Pull request by nikhiljugale007**
Added redux. Added backend from mockbee social media and made few changes to make it suitable for quiz app.
Files to check:
App.js
authSlice.js
quizSlice.js
store.js
Login.jsx
Header.jsx

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      server.logging = false;
+      users.forEach((item) =>
+        server.create("user", {
+          ...item,
+          followers: [],
+          following: [],
+          bookmarks: [],
+        })
+      );
+      // posts.forEach((item) => server.create("post", { ...item }));
```
Comment (dev-ashishsethi) [inline src/server.js]: remove commented code

**Review by dev-ashishsethi — APPROVED** [review]

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_JWT_SECRET = mysecretkey
+GENERATE_SOURCEMAP=false
```
Comment (Ankur9669) [inline .env]: fix: you should not push env file

File: src/apicalls/GetApi.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  try {
+    const response = await axios.get(url, {
+      headers: isAuthRequired ? auth : "",
+    });
+    return {
+      data: response.data,
+      success: true,
+    };
+  } catch (err) {
+    console.log(err);
```
Comment (Ankur9669) [inline src/apicalls/GetApi.js:14]: fix: can remove this console.log statement

File: src/apicalls/PostApi.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      headers: {
+        authorization: auth,
+      },
+    });
+    return {
+      data: response.data,
+      success: true,
+    };
+  } catch (err) {
+    console.log(err);
```
Comment (Ankur9669) [inline src/apicalls/PostApi.js:16]: fix: can remove clg statement

File: src/components/auth_wrapper/RequireAuth.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Navigate } from "react-router-dom";
+import { useSelector } from "react-redux";
+const RequireAuth = ({ children, from }) => {
+  const { isLoggedIn } = useSelector((store) => store.authSlice);
+  console.log(isLoggedIn);
```
Comment (Ankur9669) [inline src/components/auth_wrapper/RequireAuth.jsx]: fix: can remove console.log statement

File: src/pages/authentication/Login.jsx [hunk 6]
```diff
@@ -2,7 +2,10 @@ import "./Authentication.css";
 import { CheckboxInput, FormInput } from "../../components";
 import { Link } from "react-router-dom";
 import { useState } from "react";
-// import { useNavigate, useLocation } from "react-router";
+import { PostApi } from "../../apicalls/PostApi";
+import { useNavigate, useLocation } from "react-router";
+import { useDispatch } from "react-redux";
+import { setLoggedUser } from "../../redux/authSlice";
```
Comment (Ankur9669) [inline src/pages/authentication/Login.jsx:8]: fix: can import all files in one single file

**Review by Ankur9669 — APPROVED** [review]
The coding convetions are looking good, you can merge the code after suggested changes

File: src/server.js (hunk 1 above)
Comment (nikhiljugale007) [reply src/server.js]: done

File: src/apicalls/GetApi.js (hunk 3 above)
Comment (nikhiljugale007) [reply src/apicalls/GetApi.js:14]: I am not doing error handling now so i put that message. I will remove it when i implement error handling

File: src/components/auth_wrapper/RequireAuth.jsx (hunk 5 above)
Comment (nikhiljugale007) [reply src/components/auth_wrapper/RequireAuth.jsx]: removed

File: src/pages/authentication/Login.jsx (hunk 6 above)
Comment (nikhiljugale007) [reply src/pages/authentication/Login.jsx:8]: will implemented that.

File: .env (hunk 2 above)
Comment (nikhiljugale007) [reply .env]: removed .env file

**Pull request merged**